homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,191 @@
+
+    def __init__(self):
+        """Initialize workflow."""
+        self._user_input = {}
+        self._force_pairing = False
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        self._force_pairing = True
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            self._user_input = user_input
+            self.context["title_placeholders"] = {
+                "name": user_input.get(CONF_NAME, CONF_HOST)
+            }
+
+            return await self.async_step_pairing()
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_pairing(self, user_input=None):
+        """Display pairing form."""
+        errors = {}
+        if user_input is not None or self._force_pairing:
+            try:
+                client = await async_control_connect(self._user_input[CONF_HOST], None)
+            except PyLGTVPairException:
+                return self.async_abort(reason="error_pairing")
+            except WEBOSTV_EXCEPTIONS:
+                errors["base"] = "cannot_connect"
+            else:
+                return await self.async_step_register(self._user_input, client)
+
+        return self.async_show_form(
+            step_id="pairing",
+            errors=errors,
+        )
+
+    async def async_step_register(self, user_input, client=None):
+        """Register entity."""
+        if not client.is_registered():
+            return await self.async_step_user()
+
+        if client.client_key is None:
+            return self.async_abort(reason="client_key_not_found")
+
+        await self.async_set_unique_id(client.software_info["device_id"])
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self._user_input[CONF_HOST]}
+        )
+
+        data = {
+            CONF_HOST: self._user_input[CONF_HOST],
+            CONF_NAME: self._user_input[CONF_NAME],
+            CONF_CLIENT_SECRET: client.client_key,
+        }
+
+        return self.async_create_entry(
+            title=user_input.get(CONF_NAME, CONF_HOST), data=data
+        )
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered Webostv device."""
+        user_input = {
+            CONF_HOST: urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname,
+            CONF_NAME: discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME],
+        }
+        self.context["title_placeholders"] = {"name": user_input[CONF_NAME]}
+        self._force_pairing = True
+        return await self.async_step_user(user_input)
```

Comment:
We should probably send them to `async_step_pairing` instead and set the title place holders as well.

In `async_step_register` if is_registered is false and we started from discovery we should abort as we won't have a unique id and could generate a new config flow each time the device is discovered

Reply:
Ideally we set the unique id in this step since we can abort right away if it  already in use.  Alternatively (not preferred but sometimes there is no other option) if you only have data that you will know is unique to ssdp, you could use that and then swap it out later if you add a check to make sure there isn't an existing config entry for the ip.

Reply:
Changed config flow to use UDN from ssdp and update it to existing entries which has matching host ip